### ISSUE TITLE
[DESKTOP] - Type 'DragSelectionUpdateCallback' not found.

### ISSUE THREAD

**Issue by Biowulf21** [singerdmx/flutter-quill#1147]
I am trying Flutter-Quill version 7.0.3 on Flutter Linux (flutter doctor results below).

`[✓] Flutter (Channel master, 3.9.0-19.0.pre.58, on Fedora Linux 37
    (Workstation Edition) 6.2.8-200.fc37.x86_64, locale en_US.UTF-8)
    • Flutter version 3.9.0-19.0.pre.58 on channel master at
      /home/Biowulf21/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 9ee7148e66 (2 hours ago), 2023-03-30 02:34:22
      -0400
    • Engine revision 7dcbf0edf6
    • Dart version 3.0.0 (build 3.0.0-381.0.dev)
    • DevTools version 2.22.2`

When I import the package using `import 'package:flutter_quill/flutter_quill.dart';`, I get the following error:

```
Launching lib/main.dart on Linux in debug mode...
ERROR: ../../../.pub-cache/hosted/pub.dev/flutter_quill-7.0.3/lib/src/widgets/text_selection.dart:716:9: Error: Type 'DragSelectionUpdateCallback' not found.
ERROR:   final DragSelectionUpdateCallback? onDragSelectionUpdate;
ERROR:         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
ERROR: ../../../.pub-cache/hosted/pub.dev/flutter_quill-7.0.3/lib/src/widgets/text_selection.dart:716:9: Error: 'DragSelectionUpdateCallback' isn't a type.
ERROR:   final DragSelectionUpdateCallback? onDragSelectionUpdate;
ERROR:         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
ERROR: ../../../.pub-cache/hosted/pub.dev/flutter_quill-7.0.3/lib/src/widgets/raw_editor.dart:255:7: Error: The non-abstract class 'RawEditorState' is missing implementations for these members:
ERROR:  - TextInputClient.insertContent
ERROR: Try to either
ERROR:  - provide an implementation,
ERROR:  - inherit an implementation from a superclass or mixin,
ERROR:  - mark the class as abstract, or
ERROR:  - provide a 'noSuchMethod' implementation.
ERROR: class RawEditorState extends EditorState
ERROR:       ^^^^^^^^^^^^^^
ERROR: ../../../flutter/packages/flutter/lib/src/services/text_input.dart:1117:8: Context: 'TextInputClient.insertContent' is defined here.

ERROR:   void insertContent(KeyboardInsertedContent content) {}
ERROR:        ^^^^^^^^^^^^^
ERROR: Target kernel_snapshot failed: Exception
Exception: Build process failed
Exited
```

Also, I'm not sure if this is related but I also see an ambiguous import  error on the Text widget whenever I import the package.

**Comment by RomanIvn** [comment]
Due to incompatibility with the latest master channel of flutter. Waiting for an update aswell :(

**Comment by saconger** [comment]
Agreed. DragSelectionUpdateCallback appears to be removed from text_selection.dart in Flutter 3.10.0.

**Comment by dahef34** [comment]
Any update on this?  Flutter 3.10 is not flutter stable so no apps will compile.

**Comment by saconger** [comment]
I do not have this issue anymore; unfortunately, I can't say why. I have changed some things in my build for various reasons, so I don't know what fixed it (If it was me, flutter or quill that fixed it) I'm now running Flutter 3.13 & Quill 7.4.3.

**Comment by freshtechtips** [comment]
This bug has been fixed and we have tested building the app in Linux recently and we didn't get any build issues

if you have any questions feel free to ask us at any time.
